src/util/Collection.js

```diff
@@ -60,58 +60,110 @@ class Collection extends Map {
+    const retArr = [];
+    for (let i = 0; i < count; i++) {
+      retArr[i] = values.next().value;
+    }
+    return retArr;
   }
 
   /**
    * Obtains the first key in this collection.
-   * @returns {*}
+   * @param {number} [count] Number of keys to return. If omited, returns a single key.
+   * If present, returns an array of keys.
+   * @returns {*|Array}
    */
-  firstKey() {
-    return this.keys().next().value;
+  firstKey(count) {
+    if (!count) return this.keys().next().value;
+    if (!Number.isInteger(count)) throw new TypeError('Count must be an integer');
+    const keys = this.keys();
+    const retArr = [];
+    for (let i = 0; i < count; i++) {
+      retArr[i] = keys.next().value;
+    }
+    return retArr;
   }
 
   /**
    * Obtains the last item in this collection. This relies on the `array()` method, and thus the caching mechanism
    * applies here as well.
-   * @returns {*}
+   * @param {number} [count] Number of values to return. If omited, returns a single values.
+   * If present, returns an array of values.
+   * @returns {*|Array}
    */
-  last() {
+  last(count) {
     const arr = this.array();
-    return arr[arr.length - 1];
+    if (!count) return arr[arr.length - 1];
+    if (!Number.isInteger(count)) throw new TypeError('Count must be an integer');
+    const retArr = [];
+    const arrLen = arr.length;
+    for (let i = 0; i < count; i++) {
+      retArr[i] = arr[arrLen - (i + 1)];
+    }
+    return retArr;
   }
 
   /**
    * Obtains the last key in this collection. This relies on the `keyArray()` method, and thus the caching mechanism
    * applies here as well.
-   * @returns {*}
+   * @param {number} [count] Number of keys to return. If omited, returns a single key.
+   * If present, returns an array of keys.
+   * @returns {*|Array}
    */
-  lastKey() {
+  lastKey(count) {
     const arr = this.keyArray();
-    return arr[arr.length - 1];
+    if (!count) return arr[arr.length - 1];
+    if (!Number.isInteger(count)) throw new TypeError('Count must be an integer');
+    const retArr = [];
+    const arrLen = arr.length;
+    for (let i = 0; i < count; i++) {
+      retArr[i] = arr[arrLen - (i + 1)];
+    }
+    return retArr;
   }
 
   /**
    * Obtains a random item from this collection. This relies on the `array()` method, and thus the caching mechanism
    * applies here as well.
-   * @returns {*}
+   * @param {number} [count] Number of values to return. If omited, returns a single values.
+   * If present, returns an array of values.
+   * @returns {*|Array}
    */
-  random() {
+  random(count) {
     const arr = this.array();
-    return arr[Math.floor(Math.random() * arr.length)];
+    if (!count) return arr[Math.floor(Math.random() * arr.length)];
+    if (!Number.isInteger(count)) throw new TypeError('Count must be an integer');
+    const retArr = [];
+    for (let i = 0; i < count; i++) {
+      retArr[i] = arr[Math.floor(Math.random() * arr.length)];
+    }
+    return retArr;
```

Comment:
Rest of the PR is great, but the only issues I have are with random(count) and randomKey(count) as they could return duplicate values. To fix this, could you splice `arr` as you go along to remove items that are already set to be returned?